### DISCUSSION TITLE
What are the available values for DocumentFilter’s scheme?

### DISCUSSION THREAD

**Discussion by dsyx** [microsoft/vscode-discussions#940] [Extension Development QnA] [answered]
Hello everyone here,

I developed a number literal hoverer, see https://github.com/dsyx/vsc-number-literal.

Now, a few users have asked me to implement a feature: **enable this hoverer on the debug interface as well**. That is, hover on the VARIABLES, WATCH and other windows on the debugging interface.

I checked the documentation for `vscode.languages.registerHoverProvider` and it looks like this requires `DocumentFilter.scheme` to be implemented. But I didn't find all available values for `DocumentFilter.scheme`.

Can you help me?

**Comment by alefragnani** [comment, marked as answer]
Hi @dsyx ,

The `registerHoverProvider` works only for the _text document_ area, when you open files. The `scheme` refers to the type/origin of document you are opening, like a `file`, `remotes`, the  `search editor` and so on. It's not related to the debug view/panels.

In order to customize the debug / SCM views, you would require a different API, which I don't remember exists right now. In this case, I would suggest you to open a feature request in the VS Code repo (https://github.com/microsoft/vscode/issues/new?assignees=&labels=&projects=&template=feature_request.md), for such API.

However, if you want your hover _while debugging_ for the expressions _in the text area_, maybe the `EvaluatableExpressionProvider` (https://code.visualstudio.com/updates/v1_43#_extension-authoring) or the `InlineValuesProvider ` (https://code.visualstudio.com/updates/v1_54#_inline-value-provider-api) would worth take a look. 

Hope this helps

**Reply by dsyx** [reply]
Hi @alefragnani ,

I've filed a feature request with the vscode team: https://github.com/microsoft/vscode/issues/198983

Grateful.
